### PR TITLE
Create branch if there are commits ahead compared to target branch

### PR DESCRIPTION
**Pull request by Bene2205**
Instead of checking if the current branch ist behind or has actual file changes this check if there a commits ahead. 

In our case we have this scenario: 
```json
{
    "allChangesIncluded": true,
    "changeCounts": {},
    "changes": [],
    "commonCommit": "31743d6fe91e7678ff10b6b27aecae0712a6ec1c",
    "baseCommit": "31743d6fe91e7678ff10b6b27aecae0712a6ec1c",
    "targetCommit": "77f75073e53f30800b5efaee0a2ecb11f33ec08d",
    "aheadCount": 3,
    "behindCount": 0
}
```
So the source branch is 3 commits ahead, nothing behind and there is no changeCount, because the code file has the same change in source and target branch. 
To get the history and branches clean, it is necessary for me that the PR is created, completed and deleting the source branch. This is only possible with the check against the aheadCount.

### CODE REVIEW

**Comment by shayki5** [comment]
Hi @Bene2205, I actually changed the behavior recently because of issue #118.
So, I think it could be a task parameter, if create a PR when nothing the `behindCount` and `changeCounts` is 0.
Can you add it as a parameter please?

**Comment by Bene2205** [comment]
Sure i'll do it

File: task/task.json [hunk 1]
```diff
@@ -239,6 +239,15 @@
...
       "visibleRule": "bypassPolicy=true"
+    }, 
+    {
+      "name": "alwaysCreatePr",
+      "type": "boolean",
+      "label": "Always Create Pull Request",
+      "defaultValue": false,
+      "required": false,
+      "helpMarkDown": "If checked, a Pull Request will always be created even if there a no changed files.",
+      "visibleRule": "alwaysCreatePr=false"
```
Comment (shayki5) [inline task/task.json:250]: what is this `visibleRule`?

Comment (Bene2205) [reply task/task.json]: You're right. It's not needed.

**Review by shayki5 — APPROVED** [review]
Awesome! thank you!

**Pull request merged**